plutus-conformance/add-test-output/Spec.hs

```diff
@@ -61,58 +61,76 @@ runnerReader inp =
   , "(3) the action to run the input files through; eval (for evaluation tests) or typecheck (for typechecking tests). "
+  , "(4) whether to write output files to all inputs or only the ones missing output files."
   , "E.g. run "
-  , "`cabal run add-test-output .uplc plutus-conformance/uplc/ eval` "
+  , "`cabal run add-test-output .uplc plutus-conformance/uplc/ eval` -- --missing"
   , "to have the executable search for files with extension `.uplc` in the /uplc directory that are missing output files. "
   , " It will evaluate and create output files for them."
   ]
 
-shownEvaluationFailure :: String
-shownEvaluationFailure = "evaluation failure"
-
 main :: IO ()
 main = do
-    MkArgs extension directory run <- customExecParser (prefs showHelpOnEmpty) args
+    MkArgs extension directory run fileOpt <- customExecParser (prefs showHelpOnEmpty) args
     allInputFiles <- findByExtension [extension] directory
-    -- only choose the ones without an output file, so as to not edit the ones already with outputs
-    inputFiles <- filterM (fmap (fmap not) (\testIn -> doesFileExist (testIn <> ".expected"))) allInputFiles
+    inputFiles <-
+      case fileOpt of
+        Missing -> filterM (fmap (fmap not) (\testIn -> doesFileExist (testIn <> ".expected"))) allInputFiles
+        All     -> pure allInputFiles
     case run of
-      Eval -> do
-        for_ inputFiles
-          (\inputFile -> do
-            inputStr <- readFile inputFile
-            let parsed = runQuoteT $ UPLC.parse UPLC.program inputFile $ T.pack inputStr
-                outFilePath = inputFile <> ".expected"
-            case parsed of
-              Left (ParseErrorB peb) -> do
-                -- warn the user that the file failed to parse
-                putStrLn $ inputFile <> " failed to parse. Error written to " <> outFilePath
-                writeFile outFilePath (errorBundlePretty peb)
-              Right pro -> do
-                res <- try (evaluate $ evalUplcProg (() <$ pro)):: IO (Either SomeException (EvaluationResult UplcProg))
-                case res of
-                  Right (EvaluationSuccess prog) -> do
-                    putStrLn $ inputFile <> " evaluated; result written to " <> outFilePath
-                    writeFile outFilePath (render $ pretty prog)
-                  Right EvaluationFailure      -> do
-                    -- warn the user that the file failed to evaluate
-                    putStrLn $ inputFile <> " failed to evaluate. Failure written to " <> outFilePath
-                    writeFile outFilePath shownEvaluationFailure
-                  Left _ -> do
-                    -- warn the user that exception is thrown
-                    putStrLn $ "Exception thrown during evaluation of " <> inputFile <>".Written to " <> outFilePath
-                    writeFile outFilePath shownEvaluationFailure
-          )
+      Eval ->
+        for_ inputFiles printAndWrite
+          where printAndWrite inputFile = do
```

Comment:
I'd write it as `for_ inputFiles $ \inputFile -> do`, but this way it's also fine.